### PR TITLE
Add confs for metric prefix and metric dataset info to mlflow.evaluate(), rename accuracy to accuracy_score

### PR DESCRIPTION
**Pull request by dbczumar**
## What changes are proposed in this pull request?

Add confs for metric prefix and metric dataset info to mlflow.evaluate(), rename accuracy to accuracy_score

## How is this patch tested?

Added integration tests

## Does this PR change the documentation?

- [X] No. You can skip the rest of this section.
- [ ] Yes. Make sure the changed pages / sections render correctly by following the steps below.

1. Click the `Details` link on the `Preview docs` check.
2. Find the changed pages / sections and make sure they render correctly.

## Release Notes

Add confs for metric prefix and metric dataset info to mlflow.evaluate(), rename accuracy to accuracy_score

### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [X] Yes. Give a description of this change to be included in the release notes for MLflow users.

### What component(s), interfaces, languages, and integrations does this PR affect?
Components
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [X] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/pipelines`: Pipelines, Pipeline APIs, Pipeline configs, Pipeline Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface
- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>
### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [X] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: mlflow/models/evaluation/default_evaluator.py [hunk 1]
```diff
@@ -175,7 +171,7 @@ def _get_classifier_global_metrics(is_binomial, y, y_pred, y_probs, labels):
     get classifier metrics which computing over all classes examples.
     """
     metrics = {}
-    metrics["accuracy"] = sk_metrics.accuracy_score(y, y_pred)
+    metrics["accuracy_score"] = sk_metrics.accuracy_score(y, y_pred)
```
Comment (dbczumar) [inline mlflow/models/evaluation/default_evaluator.py:174]: @yxiong @jimmyxu-db Renaming for consistency with `mlflow.sklearn.eval_and_log_metrics`; let me know if this is unnecessary

File: tests/models/test_default_evaluator.py [hunk 2]
```diff
@@ -1173,3 +1173,41 @@ def add_feature(df):
...
+            model_info.model_uri,
+            X.assign(target=y),
+            model_type="classifier" if isinstance(model, LogisticRegression) else "regressor",
+            targets="target",
+            dataset_name="iris",
+            evaluators="default",
+            evaluator_config={
+                "metric_prefix": prefix,
+                "log_metrics_with_dataset_info": log_metrics_with_dataset_info,
+            },
```
Comment (dbczumar) [inline tests/models/test_default_evaluator.py:1196]: @yxiong @jimmyxu-db Here's an example of how the new config arguments can be used to set a prefix and disable metric logging with dataset info

File: mlflow/models/evaluation/default_evaluator.py (hunk 1 above)
Comment (yxiong) [reply mlflow/models/evaluation/default_evaluator.py:174]: I think it's nice to be consistent :)

**Review by yxiong — APPROVED** [review]

**Review by jimmyxu-db — APPROVED** [review]
Thanks for the quick PR!
